_tests/test_endpoints.py

```diff
@@ -0,0 +1,92 @@
+# -*- coding: utf-8 -*-
+import requests
+import pytest
+from collections import namedtuple
+import glob
+import os
+import json
+import logging
+
+# Setup basic configuration for logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+EndpointTest = namedtuple('EndpointTest', ['chain', 'endpoint', 'provider', 'address'])
+
+# Set this to False to ignore the whitelist and process all providers
+use_whitelist = True
+
+# Whitelist for specific chains and providers
+whitelist = {
+    "chains": [
+        "axelar",
+        "celestia",
+        "composable",
+        "cosmoshub",
+        "dydx",
+        "dymension",
+        "evmos",
+        "injective",
+        "neutron",
+        "noble",
+        "osmosis",
+        "stargaze",
+        "stride"
+    ],
+    "providers": [
+        "Osmosis Foundation", 
+        "Polkachu",
+        "CryptoCrew",
+        "forbole",
+        "Imperator.co",
+        "WhisperNode 🤐",
+        "chainlayer",
+        "Numia",
+        "Enigma",
+        "kjnodes",
+        "Stake&Relax 🦥",
+        "Allnodes ⚡️ Nodes & Staking",
+        "Lava",
+        "Golden Ratio Staking",
+        "Stargaze Foundation",
+    ]
+} if use_whitelist else {'chains': [], 'providers': []}
+
+def log_request_details(test_case, response):
+    logging.info(f"Testing {test_case.chain}-{test_case.endpoint.upper()}-{test_case.provider}")
+    logging.info(f"Request URL: {response.url}")
+    logging.info(f"Response Status Code: {response.status_code}")
+    logging.info(f"Response Body: {response.text}")
+
+def generate_endpoint_tests():
+    test_cases = []
+    logging.info(f"Current working directory: {os.getcwd()}")
+    files_found = glob.glob('**/chain.json', recursive=True)
+    for filename in files_found:
+        with open(filename) as f:
+            data = json.load(f)
+            chain_name = data.get('chain_name', 'unknown')
+            if 'apis' in data:
```

Comment:
Might be helpful to print some warnings if this condition is false in case the structure changes and we start skipping the entries.

Applies to other if statements below too

Reply:
Added